src/SelfServe/SqlX/SqlX.rp.ts

```diff
@@ -1,33 +1,89 @@
 import { RefreshResult } from "../SelfServeTypes";
+import { userContext } from "../../UserContext";
+import { armRequest } from "../../Utils/arm/request";
+import { configContext } from "../../ConfigContext";
+import {
+  SqlxServiceResource,
+  UpdateDedicatedGatewayRequestParameters
+} from "./SqlxTypes"
+
+const apiVersion = "2020-06-01-preview";
 
 export interface DedicatedGatewayResponse {
   sku: string;
   instances: number;
+  status: string;
+  endpoint: string;
 }
 
-export const getRegionSpecificMinInstances = async (): Promise<number> => {
-  // TODO: write RP call to get min number of instances needed for this region
-  throw new Error("getRegionSpecificMinInstances not implemented");
-};
-
-export const getRegionSpecificMaxInstances = async (): Promise<number> => {
-  // TODO: write RP call to get max number of instances needed for this region
-  throw new Error("getRegionSpecificMaxInstances not implemented");
-};
+export enum SKU {
+  CosmosD4s = "Cosmos.D4s",
+  CosmosD8s = "Cosmos.D8s",
+  CosmosD16s = "Cosmos.D16s",
+  CosmosD32s = "Cosmos.D32s"
+}
 
-export const updateDedicatedGatewayProvisioning = async (sku: string, instances: number): Promise<void> => {
+export const updateDedicatedGatewayResource = async (sku: string, instances: number): Promise<void> => {
   // TODO: write RP call to update dedicated gateway provisioning
-  throw new Error(
-    `updateDedicatedGatewayProvisioning not implemented. Parameters- sku: ${sku}, instances:${instances}`
-  );
+  const path = `/subscriptions/${userContext.subscriptionId}/resourceGroups/${userContext.resourceGroup}/providers/Microsoft.DocumentDB/databaseAccounts/${userContext.databaseAccount.name}/services/sqlx`;
```

Comment:
can you make a getPath() function that takes subscriptionIf, resourceGroup name and databaseAccount name, looks like it is reused quite a bit

Reply:
Okay, I don't really think that is necessary, it does not really achieve a lot. But I will make a change for this iteration.